docs/general/Mainnet/node_level_architecture.md

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 2
+---
+
+# Node-level architecture
+
+## Node Types
+
+### Full node
+
+A full node provides the entire functionality of a Movement node:
+
+- ingresses transactions into a local mempool
+- forms proto-blocks from the transactions in the mempool and forwards them
+  to the [DA light node](#da-light-node)
+- runs executions of blocks sequenced by the DA,
+  receiving them through the DA light node.
+- provides views on the optimistic and settled (finalized) states via
+  Aptos RPC API endpoints.
+- (in a future decentralized network) can operate as a [validator node](#validator-node) with a key from the trusted validator set.
+
+### Validator node
+
+A validator node is a full node that is part of the trusted validator set which settles blocks via the [Fast-Finality-Settlement](high_level_architecture.md#settlement).
+
+### Follower Node
+
+The Movement [Follower Node][follower] is a node type introduced for RPC providers. A follower node
+performs most of the functions provided by the full node, except settlement of blocks.
```

Comment:
Isn't it the validator that settles blocks?

Reply:
yes. @mzabaluev i see you added this. how does the follower node differ from the full node?